src/source.js

```diff
@@ -34,6 +34,7 @@ async function openS3Stream (bucketRegion, url, retries = s3MaxRetries, retryDel
     try {
       // this imports just the getObject operation from S3
       s3Request = await telemetry.trackDuration('s3-fetchs', s3Client.send(new GetObjectCommand({ Bucket, Key })))
+      break
```

Comment:
I think this should also be added in `storage.js`

Reply:
Actually if we're going to keep that in the code we should centralize that retry logic somewhere, if possible, to avoid changing in one place and forgetting about the other. Duplicated code in `source.js` and `storage.js`